### PR TITLE
feat: unlimited intro dialog

### PR DESCRIPTION
**Pull request by moo-onthelawn**
figma [mock](https://www.figma.com/design/fMQodZKzn9B5aZTN5G0fLJ/dYdX-%E2%80%BA-Web?node-id=42754-198546&node-type=frame&t=M7rNeq0JETNG8Cdo-0).

1. Create the Unlimited Announcement dialog sequence
2. Small update to remove border from `Primary` buttons (make it match background) to match Figmas

TODO:
- [x] get smaller svg for `unlimited-affiliates.svg` and `unlimited-megavault.svg` from achal (slow to load)
- [x] add graphics from achal

---

## Views

- `<App>`
  - When wallet is connected, and if user has not yet dismissed the announcement, show the Unlimited Dialog
  - Modify referral code dialog to only pop up if no active dialogs open (I think dev is still occurring here and there will probably be another condition for rendering)

- `<HeaderDesktop>`
  - Nit bug fix that prevents deposit button from flashing from purple -> grey on most accounts

- `<dialogs/UnlimitedAnnouncementDialog/...>`
  - Create Dialog and dialog components

## Components

- `<Dialog>`
  - Add a `slotHeader` prop

- `<Button>`
  - Update Primary button border to be `Accent`

## Styles/Mixins

- `styles/test.css`
  - Add `medium` `bold` font

## Constants/Types

- `constants/statsig.ts`
  - Add `ffShowUnlimitedAnnouncement` flag (currently enabled for dev environments)

- `constants/dialogs.ts`
  - Add `UnlimitedAnnouncement` dialog type
 
- new: `constants/unlimitedAnnouncement.ts`
  - Add steps to `UnlimitedAnnouncement` dialog
  
## Functions/Clients

- `lib/_____`
  - 
  -

## State

- `state/dismissable`
  - Add `hasSeenUnlimitedAnnouncement` to state and setter

- `state/dismissableSelectors`
  - Add getter for `hasSeenUnlimitedAnnouncement`

## Packages

- `v4-localization`

---

### CODE REVIEW

File: src/styles/text.css [hunk 1]
```diff
@@ -45,6 +45,7 @@
   --font-medium-regular: var(--fontWeight-regular) var(--fontSize-medium) var(--fontFamily-base);
   --font-medium-book: var(--fontWeight-book) var(--fontSize-medium) var(--fontFamily-base);
   --font-medium-medium: var(--fontWeight-medium) var(--fontSize-medium) var(--fontFamily-base);
+  --font-medium-bold: var(--fontWeight-bold) var(--fontSize-medium) var(--fontFamily-base);
```
Comment (tyleroooo) [inline src/styles/text.css:48]: should also add to tailwind config

File: src/state/migrations/2.ts [hunk 2]
```diff
@@ -24,6 +24,7 @@ export function migration2(state: PersistedState | undefined): V2State {
     dismissable: {
       hasSeenPredictionMarketIntroDialog: hasSeenPredictionMarketsIntro ?? false,
       dismissedAffiliateBanner: false,
+      hasSeenUnlimitedAnnouncement: false,
```
Comment (tyleroooo) [inline src/state/migrations/2.ts:27]: I don't totally understand why we are modifying old migrations. Is it really necessary?

File: src/constants/unlimitedAnnouncement.ts [hunk 3]
```diff
@@ -0,0 +1,9 @@
+export enum UnlimitedAnnouncementDialogSteps {
+  Announcement = 'Announcement',
+  MarketListings = 'MarketListings',
+  MegaVault = 'MegaVault',
+  AffiliatesProgram = 'Affiliates',
+  Incentives = 'Incentives',
+}
+
+export const unlimitedAnnouncementStepOrder = Object.values(UnlimitedAnnouncementDialogSteps);
```
Comment (tyleroooo) [inline src/constants/unlimitedAnnouncement.ts:9]: lol is this safe? I'm not sure we are guaranteed key ordering with normal objects.

File: src/components/Dialog.tsx [hunk 4]
```diff
@@ -163,7 +174,7 @@ export const Dialog = ({
             </$Header>
           )}
 
-          <$Content>{children}</$Content>
+          <$Content> {children}</$Content>
```
Comment (tyleroooo) [inline src/components/Dialog.tsx:177]: intentional?

**Comment by tyleroooo** [comment]
Code ready, blocked on Achal getting back to me.

File: src/views/dialogs/UnlimitedAnnouncementDialog/UnlimitedAnnouncementHeader.tsx [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+  border: 3px solid var(--color-border);
+  border-radius: 1rem;
+  padding: 0.25rem 0.5rem;
+
+  color: var(--color-text-2);
+  font: var(--font-extra-medium);
+`;
+
+const $PlaceholderTitle = styled.div`
+  height: 360px;
```
Comment (jaredvu) [inline src/views/dialogs/UnlimitedAnnouncementDialog/UnlimitedAnnouncementHeader.tsx:85]: px -> rem

File: src/views/dialogs/UnlimitedAnnouncementDialog/UnlimitedAnnouncementHeader.tsx [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+            tw="absolute h-auto w-full"
+            alt=""
+          />
+        </$PlaceholderTitle>
+      );
+  }
+};
+
+const $Title = styled.div`
+  height: 200px;
```
Comment (jaredvu) [inline src/views/dialogs/UnlimitedAnnouncementDialog/UnlimitedAnnouncementHeader.tsx:63]: px -> rem

**Review by jaredvu — APPROVED** [review]

**Pull request merged**